cpp/src/arrow/util/meson.build

```
@@ -92,3 +92,243 @@ configure_file(
+        'aligned_storage.h',
+        'align_util.h',
+        'async_generator_fwd.h',
+        'async_generator.h',
+        'async_util.h',
+        'base64.h',
+        'basic_decimal.h',
+        'benchmark_util.h',
+        'binary_view_util.h',
+        'bit_block_counter.h',
+        'bitmap_builders.h',
+        'bitmap_generate.h',
+        'bitmap.h',
+        'bitmap_ops.h',
+        'bitmap_reader.h',
+        'bitmap_visit.h',
+        'bitmap_writer.h',
+        'bit_run_reader.h',
+        'bitset_stack.h',
+        'bit_util.h',
+        'bpacking64_default.h',
+        'bpacking_avx2.h',
+        'bpacking_avx512.h',
+        'bpacking_default.h',
+        'bpacking.h',
+        'bpacking_neon.h',
+        'byte_size.h',
+        'cancel.h',
+        'checked_cast.h',
+        'compare.h',
+        'compression.h',
+        'concurrent_map.h',
+        'converter.h',
+        'counting_semaphore.h',
+        'cpu_info.h',
+        'crc32.h',
+        'debug.h',
+        'decimal.h',
+        'delimiting.h',
+        'dict_util.h',
+        'dispatch.h',
+        'double_conversion.h',
+        'endian.h',
+        'float16.h',
+        'formatting.h',
+        'functional.h',
+        'future.h',
+        'hashing.h',
+        'hash_util.h',
+        'int_util.h',
+        'int_util_overflow.h',
+        'io_util.h',
+        'iterator.h',
+        'key_value_metadata.h',
+        'launder.h',
+        'list_util.h',
+        'logger.h',
+        'logging.h',
+        'macros.h',
+        'map.h',
+        'math_constants.h',
+        'memory.h',
+        'mutex.h',
+        'parallel.h',
+        'pcg_random.h',
+        'prefetch.h',
+        'print.h',
+        'queue.h',
+        'range.h',
+        'ree_util.h',
+        'regex.h',
+        'rows_to_batches.h',
+        'simd.h',
+        'small_vector.h',
+        'sort.h',
+        'spaced.h',
+        'span.h',
+        'stopwatch.h',
+        'string_builder.h',
+        'string.h',
+        'task_group.h',
+        'tdigest.h',
+        'test_common.h',
+        'thread_pool.h',
+        'time.h',
+        'tracing.h',
+        'trie.h',
+        'type_fwd.h',
+        'type_traits.h',
+        'ubsan.h',
+        'union_util.h',
+        'unreachable.h',
+        'uri.h',
+        'utf8.h',
+        'value_parsing.h',
+        'vector.h',
+        'visibility.h',
+        'windows_compatibility.h',
+        'windows_fixup.h',
+    ],
+    subdir: 'arrow/util',
+)
+
+if host_machine.system() == 'windows'
+    # This manifest enables long file paths on Windows 10+
+    # See https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#enable-long-paths-in-windows-10-version-1607-and-later
+    if cpp_compiler.get_id() == 'msvc'
+        io_util_test_sources = ['io_util_test.cc', 'io_util_test.manifest']
+    else
+        io_util_test_sources = ['io_util_test.cc', 'io_util_test.rc']
+    endif
+else
+    io_util_test_sources = ['io_util_test.cc']
+endif
```

```suggestion
io_util_test_sources = ['io_util_test.cc']
if host_machine.system() == 'windows'
    # This manifest enables long file paths on Windows 10+
    # See https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#enable-long-paths-in-windows-10-version-1607-and-later
    if cpp_compiler.get_id() == 'msvc'
        io_util_test_sources += ['io_util_test.manifest']
    else
        io_util_test_sources += ['io_util_test.rc']
    endif
endif
```

Can we use the following?

```meson
utility_test_srcs = [
    ...,
    'io_util_test.cc'
    ...
]
if host_machine.system() == 'windows'
    # This manifest enables long file paths on Windows 10+
    # See https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#enable-long-paths-in-windows-10-version-1607-and-later
    if cpp_compiler.get_id() == 'msvc'
        utility_test_srcs += ['io_util_test.manifest']
    else
        utility_test_srcs += ['io_util_test.rc']
    endif
endif